src/Compiler/SyntaxTree/SyntaxTree.fs

```diff
@@ -306,7 +306,23 @@ type DebugPointAtBinding =
 
 type SeqExprOnly = SeqExprOnly of bool
 
-type BlockSeparator = range * pos option
+[<NoEquality; NoComparison; RequireQualifiedAccess>]
+type BlockSeparator =
+    | Semicolon of range: range * position: pos option
+    | Comma of range: range * position: pos option
+    | Offside of range: range * position: pos option
```

Comment:
@edgarfgp I don't understand how these types are supposed to work in practice. The separator is wrapped in an `option` in other places in the tree, e.g. in `NamePatPairField`. Let's assume the value is Some, i.e. there is a separator. How could it ever be that the position is `None` in such case?

Reply:
Let me create a PR to make Semicolon position no optional